### DISCUSSION TITLE
I have ESP32 wiring diagram

### DISCUSSION THREAD

**Discussion by harinderbhinder** [grblHAL/ESP32#108] [Q&A]
I have ESP32(38pins) and I want to connect 3 motors and limit switches (X, Y, Z) I want a wiring diagram
I don't have a CNC board I just want to connect directly to esp32 board

**Comment by terjeio** [comment]
[Some boards](http://svn.io-engineering.com:8080/?driver=ESP32) has links to the schematic, or just create one from the board map. You should add signal conditioning for the inputs, I prefer optocouplers - add a RC network as a minimum.

**Reply by harinderbhinder** [reply]
![73596140-8ac2d000-44d4-11ea-8025-567c5fda3be0](https://github.com/grblHAL/ESP32/assets/5358099/7eee69c1-e192-4507-b36c-fa8b9d24520c)
@terjeio I found this can you please check if its correct or not?
